### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by alexashley**
Just policy group CRUD to start. Next, I'd like to test the policies, assignments, and authn/authz.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -25,6 +25,39 @@ jobs:
...
+          name: rode
+
+      - name: Restore File Permissions
+        run: |
+          chmod +x ./rode # permissions are stripped by artifact upload
+
+      - name: Start Rode Environment
+        uses: rode/rode-dev-env@v0.1.2
+        with:
+          rodeVersion: 'local'
```
Comment (alexashley) [inline .github/workflows/test.yml:56]: With `rodeVersion` set to `local` the action will start the Rode binary, wait for it be ready, and then, at the end, stop it and collect its logs.

**Review by RobertKelly — APPROVED** [review]
Nice to see the integration tests

File: test/v1alpha1/policy_groups_test.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+	Describe("Creating a policy group", func() {
+		When("the policy group name is valid", func() {
+			It("should create the policy group", func() {
+				expectedName := strings.ToLower(fake.LetterN(10))
+				_, err := rode.CreatePolicyGroup(ctx, &v1alpha1.PolicyGroup{
+					Name: expectedName,
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				group, err := rode.GetPolicyGroup(ctx, &v1alpha1.GetPolicyGroupRequest{Name: expectedName})
```
Comment (mrparkers) [inline test/v1alpha1/policy_groups_test.go:42]: nitpick: we could use the return value here as a way to assert that the policy group that was returned has the expected name

```suggestion
				createdPolicyGroup, err := rode.CreatePolicyGroup(ctx, &v1alpha1.PolicyGroup{
					Name: expectedName,
				})
				Expect(err).NotTo(HaveOccurred())

				group, err := rode.GetPolicyGroup(ctx, &v1alpha1.GetPolicyGroupRequest{Name: createdPolicyGroup.Name})
```

File: test/v1alpha1/policy_groups_test.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+			})
+		})
+	})
+
+	Describe("Updating a policy group", func() {
+		When("the description has changed", func() {
+			It("should be updated", func() {
+				expectedName := strings.ToLower(fake.LetterN(10))
+
+				_, err := rode.CreatePolicyGroup(ctx, &v1alpha1.PolicyGroup{Name: expectedName})
```
Comment (mrparkers) [inline test/v1alpha1/policy_groups_test.go:113]: same nitpick here

**Review by mrparkers — APPROVED** [review]

**Pull request merged**